src/v2/api/services/RoleService.ts

```diff
@@ -103,4 +107,77 @@ export class RoleService {
 
     await this.checkRole(userRoles, parent);
   }
-}
\ No newline at end of file
+
+
+  async transferRole (captainUserId : string, studentId: string) {
+    const { id: groupId } = await this.groupRepository.find({
+      students: {
+        some: {
+          userId: {
+            in: [captainUserId, studentId],
```

Comment:
This check doesn't determine whether these students are in the same group or not. You need to change request so both students are in the same group.

As you can see, `in` operator is `OR` logical operator underhood. That means that you can't realise `AND` operation in a such way (and means that group should contain one AND another student). So you must split this `some` operator into two.